packages/next/src/client/components/layout-router.tsx

```diff
@@ -295,10 +295,7 @@ class InnerScrollAndFocusHandler extends React.Component<ScrollAndFocusHandlerPr
   }
 
   componentDidUpdate() {
-    // Because this property is overwritten in handlePotentialScroll it's fine to always run it when true as it'll be set to false for subsequent renders.
-    if (this.props.focusAndScrollRef.apply) {
```

Comment:
We already check this within `handlePotentialScroll`. New factoring makes it clearer that we can convert this to `useLayoutEffect`